backend/app/services/aiod.py

```diff
@@ -0,0 +1,164 @@
+from enum import Enum
+from pathlib import Path
+from urllib.parse import urljoin
+
+import httpx
+from fastapi import HTTPException
+from pydantic import Json
+
+from app.config import settings
+from app.helpers import Pagination
+from app.schemas.dataset import Dataset
+from app.schemas.ml_model import MLModel
+
+
+class AssetType(Enum):
+    DATASETS: str = "datasets"
+    ML_MODELS: str = "ml_models"
+    PUBLICATIONS: str = "publications"
+    PLATFORMS: str = "platforms"
+
+
+class AIoDClientWrapper:
```

Comment:
What's the benefit of this wrapper compared to if we only created a client on app startup (and shat it down on app shutdown)?
Is it because we want to mock the client in tests?

It forces you to access the client through "aiod_client_wrapper.client" instead of just "client".

Reply:
To be honest, I am not completely sure at the moment, since this was done some longer time back. As for this PR, I didn't want to make any major modification to this. We can look into it later in a separate refactoring if we want.
There are definitely some benefits when doing it this way compared to what you are suggesting, but I feel there is another option somewhere in between. Nevertheless, I decided to not do it in this PR.